symforce/slam/imu_preintegration/imu_factor.cc

```diff
@@ -0,0 +1,38 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#include "./imu_factor.h"
+
+#include <Eigen/Cholesky>
+
+#include <sym/factors/internal_imu_factor.h>
+
+namespace sym {
+
+template <typename Scalar>
+ImuFactor<Scalar>::ImuFactor(const PreintegratedImuMeasurements<Scalar>& pim)
+    : pim_{pim.pim},
+      sqrt_info_{pim.covariance.llt().matrixL().solve(Eigen::Matrix<Scalar, 9, 9>::Identity())} {}
```

Comment:
Can we add a comment saying that `sqrt_info_` is now lower triangular?
And a quick explanation on why we do this? 
Also
```
pim.covariance.selfAdjointView<Eigen::Lower>().llt()
```

Reply:
Good idea to mention it's lower triangular. And a comment to explain why `llt()` then invert rather than the other way around?

And does it actually make a difference to use the `selfAdjointView`? It certainly seems plausible, but I would be kind of surprised considering we always have to assume the matrix is self-adjoint in order to perform the cholesky decomposition anyway.

From the docs:
[Note that during the decomposition, only the lower (or upper, as defined by UpLo_) triangular part of A is considered. Therefore, the strict lower part does not have to store correct values.](https://eigen.tuxfamily.org/dox/classEigen_1_1LLT.html#:~:text=Note%20that%20during%20the%20decomposition%2C%20only%20the%20lower%20(or%20upper%2C%20as%20defined%20by%20UpLo_)%20triangular%20part%20of%20A%20is%20considered.%20Therefore%2C%20the%20strict%20lower%20part%20does%20not%20have%20to%20store%20correct%20values.)

Then, looking at the definitions of the respective `llt` methods (at the bottom of https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Cholesky/LLT.h)

```cpp
/** \cholesky_module
  * \returns the LLT decomposition of \c *this
  * \sa SelfAdjointView::llt()
  */
template<typename Derived>
inline const LLT<typename MatrixBase<Derived>::PlainObject>
MatrixBase<Derived>::llt() const
{
  return LLT<PlainObject>(derived());
}

/** \cholesky_module
  * \returns the LLT decomposition of \c *this
  * \sa SelfAdjointView::llt()
  */
template<typename MatrixType, unsigned int UpLo>
inline const LLT<typename SelfAdjointView<MatrixType, UpLo>::PlainObject, UpLo>
SelfAdjointView<MatrixType, UpLo>::llt() const
{
  return LLT<PlainObject,UpLo>(m_matrix);
}
```

So it seems like that comment (about not needing the strict lower/upper part) applies to both versions.

Reply:
Yep agree the LLT only reads one triangle so it shouldn't make a difference whether you take a selfAdjointView or not